addons/mail/models/mail_blacklist.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class MailBlackList(models.Model):
+    """ Model of blacklisted email addresses to stop sending emails."""
+    _name = 'mail.blacklist'
+    _inherit = ['mail.thread']
+    _description = 'Mail Blacklist'
+    _rec_name = 'email'
+
+    email = fields.Char(string='Email Address', required=True, index=True, help='This field is case insensitive.',
+                        track_visibility='always')
+    active = fields.Boolean(default=True, track_visibility='onchange')
+
+    _sql_constraints = [
+        ('unique_email', 'unique (email)', 'Email address already exists!')
+    ]
+
+    @api.model
+    def create(self, values):
+        """ To avoid crash during import due to unique email, return the existing record if any """
+        rec = self.search([('email', '=ilike', values['email'])])
+        if len(rec) == 1:
+            return rec
+        return super(MailBlackList, self).create(values)
+
+    def _add(self, email):
+        blacklist = self.env["mail.blacklist"].sudo()
+        record = blacklist.with_context(active_test=False).search([('email', '=', email)])
+        if len(record) > 0:
+            record.active = True
+        else:
+            record = self.create({'email': email})
+        return record
+
+    def _remove(self, email):
+        blacklist = self.env["mail.blacklist"].sudo()
```

Comment:
As convention, use a capital letter when using the variable as the 'model' to call method. And you can suffix it by `_sudo` so we know this is done as sudo. So, `Blacklist_sudo`.
After all, it might not be required to use a variable here ... ^^'